8196587: Remove use of deprecated finalize method from JPEGImageLoader

arapte opened the pull request:
The finalize() method is deprecated in JDK9. See [Java 9 deprecated features](https://www.oracle.com/technetwork/java/javase/9-deprecated-features-3745636.html).
And so the JPEGImageLoader.finalize() method should be removed.

The change is,
1. Remove finalize method from JPEGImageLoader class.

2. Instance of JPEGImageLoader is created and used in ImageStorage class. JPEGImageLoader.dispose() should be called after it's use over. This would be a common call for the other (GIF, PNG, BMP) ImageLoader classes, which have empty dispose() method.

3. JPEGImageLoader.load() method almost always calls the dispose() method after the image is loaded. In normal scenario JPEGImageLoader is disposed here. The calls to dispose() added in ImageStorage seem logically correct place to add and should be added.

Verification:
Verified :graphics:test and system tests on all three platforms.
Verified that JPEGImageLoader.dispose() is always initiated by JPEGImageLoader.load()
<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
## Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

## Issue
[JDK-8196587](https://bugs.openjdk.java.net/browse/JDK-8196587): Remove use of deprecated finalize method from JPEGImageLoader


## Approvers
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)
 * Johan Vos ([jvos](@johanvos) - **Reviewer**)

kevinrushforth commented on modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java:
```diff
@@ -265,17 +267,18 @@ public static void addImageLoaderFactory(ImageLoaderFactory factory) {
+            if (loader != null) {
+                images = loadAll(loader, width, height, preserveAspectRatio, pixelScale, smooth);
+            } else {
+                throw new ImageStorageException("No loader for image data");
```
Now that this is moved inside a try/catch, this `ImageStorageException` will get wrapped in another `ImageStorageException` if it is ever thrown. You probably want to catch `ImageStorageException` below and re-throw it without wrapping it.

arapte replied:
I will change the catch as below and update with next commit.
```
} catch (ImageStorageException ise) {
        throw ise;
} catch (IOException e) {
        throw new ImageStorageException(e.getMessage(), e);
} 
```

arapte replied:
Updated the catch statement.

kevinrushforth commented:
There is one more ImageLoader implementation class that still has a finalize method: [IosImageLoader.java(https://github.com/openjdk/jfx/blob/3ba659f0eb52496adb54a2b2c313e8da832e8389/modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ios/IosImageLoader.java#L215).

That method should also be removed, but @johanvos should confirm.

The rest looks fine to me.

kevinrushforth commented:
This will need two reviewers anyway, so I tagged @johanvos to review as well.

arapte commented:
> There is one more ImageLoader implementation class that still has a finalize method: [IosImageLoader.java](https://github.com/openjdk/jfx/blob/3ba659f0eb52496adb54a2b2c313e8da832e8389/modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ios/IosImageLoader.java#L215).

Removed `finalize()` from `IosImageLoader.java`, updated PR.

arapte commented:
/integrate
